packages/core/src/checkout-buttons/strategies/braintree/braintree-paypal-credit-button-strategy.ts

```diff
@@ -37,15 +44,30 @@ export default class BraintreePaypalCreditButtonStrategy implements CheckoutButt
             throw new InvalidArgumentError(`Unable to initialize payment because "options.braintreepaypalcredit" argument is not provided.`);
         }
 
-        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
-        const currency = state.cart.getCartOrThrow()?.currency.code;
+        let state: InternalCheckoutSelectors;
+        let paymentMethod: PaymentMethod;
+        let currencyCode: string;
+
+        if (braintreepaypalcredit?.buyNowInitializeOptions) {
```

Comment:
Do you need a `?` here? We already check on line 42 of the property is undefined or not.

Reply:
@animesh1987 
removed this check